### PR TITLE
Add deprecation note for Gatekeeper in generator

### PR DESCRIPTION
**Pull request by dhaiducek**
Also adds deprecation notes for Placement in the generator.

ref: https://issues.redhat.com/browse/ACM-5160

### CODE REVIEW

**Review by mprahl — APPROVED** [review]

File: governance/policy_generator.adoc [hunk 1]
```diff
@@ -279,7 +279,9 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 
 | `policyDefaults.informGatekeeperPolicies`
```
Comment (dockerymick) [inline governance/policy_generator.adoc]: I see I am using "configuration policy" throughout the doc. If we refer to the `kind` value of the policy, it is monospaced. The plus sign is added 
```suggestion
| `policyDefaults.informGatekeeperPolicies` (Deprecated)
| Optional
| Set `informGatekeeperPolicies` to false to use Gatekeeper manifests directly without defining it in a configuration policy.

When the policy references a violated Gatekeeper policy manifest, an additional configuration policy is generated in order to receive policy violations in {product-title-short}. The default value is `true`.

```

File: governance/policy_generator.adoc [hunk 2]
```diff
@@ -315,15 +317,21 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 
 | `policyDefaults.placement.clusterSelector`
```
Comment (dockerymick) [inline governance/policy_generator.adoc]:
```suggestion
| `policyDefaults.placement.clusterSelector` (Deprecated)
| Optional
| `PlacementRule` is deprecated. Use `labelSelector` instead to generate a placement. 

Specify a placement by defining a cluster selector using either `key:value` or by providing `matchExpressions`, `matchLabels`, or both, with appropriate values. See `placementRulePath` to specify an existing file.

```

File: governance/policy_generator.adoc [hunk 3]
```diff
@@ -315,15 +317,21 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 | Optional
-| Specify a placement by defining a cluster selector using either `key:value`, or by providing `matchExpressions`, `matchLabels`, or both, with appropriate values. See `_placementRulePath_` to specify an existing file.
+| Deprecated: PlacementRule is deprecated. Use `labelSelector` instead to generate a Placement. 
+
+Specify a placement by defining a cluster selector using either `key:value`, or by providing `matchExpressions`, `matchLabels`, or both, with appropriate values. See `_placementRulePath_` to specify an existing file.
 
 | `policyDefaults.placement.placementRuleName`
```
Comment (dockerymick) [inline governance/policy_generator.adoc]:
```suggestion
| `policyDefaults.placement.placementRuleName` (Deprecated)
| Optional
| `PlacementRule` is deprecated. Alternatively, use `placementName` to specify a placement.

To use an existing placement rule on the cluster, specify the name for this parameter. A `PlacementRule` is not created, but a `PlacementBinding` binds the policy to the existing `PlacementRule`.

```

File: governance/policy_generator.adoc [hunk 4]
```diff
@@ -315,15 +317,21 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 | Optional
-| To use a placement rule that already exists on the cluster, specify its name here. A `PlacementRule` is not created, but a `PlacementBinding` binds the policy to this `PlacementRule`.
+| Deprecated: PlacementRule is deprecated. Use `placementName` instead to specify a Placement.
+    
+To use a placement rule that already exists on the cluster, specify its name here. A `PlacementRule` is not created, but a `PlacementBinding` binds the policy to this `PlacementRule`.
 
 | `policyDefaults.placement.placementRulePath`
```
Comment (dockerymick) [inline governance/policy_generator.adoc]:
```suggestion
| `policyDefaults.placement.placementRulePath` (Deprecated)
| Optional
| `PlacementRule` is deprecated. Alternatively, use `placementPath` to specify a placement.

To reuse an existing placement rule, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_clusterSelector_` to generate a new `PlacementRule`.

```

**Review by dockerymick — CHANGES_REQUESTED** [review]

**Comment by dockerymick** [comment]
@dhaiducek i made some updates to the PR. When you have an opportunity, will you review the PR? If you are fine with the updates, I will approve and merge

File: governance/policy_generator.adoc [hunk 5]
```diff
@@ -315,23 +313,17 @@ When the policy references a violated gatekeeper policy manifest, this determine
 | Optional
 | To reuse an existing placement, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_labelSelector_` to generate a new `Placement`.
 
-| `policyDefaults.placement.clusterSelector`
+| `policyDefaults.placement.clusterSelector` (Deprecated)
 | Optional
-| Deprecated: PlacementRule is deprecated. Use `labelSelector` instead to generate a Placement. 
-
-Specify a placement by defining a cluster selector using either `key:value`, or by providing `matchExpressions`, `matchLabels`, or both, with appropriate values. See `_placementRulePath_` to specify an existing file.
+| `PlacementRule` is deprecated. Use `labelSelector` instead to generate a placement. Specify a placement by defining a cluster selector using either `key:value` or by providing `matchExpressions`, `matchLabels`, or both, with appropriate values. See `placementRulePath` to specify an existing file.
```
Comment (dhaiducek) [inline governance/policy_generator.adoc]:
```suggestion
| `PlacementRule` is deprecated. Use `labelSelector` instead to generate a placement. Specify a placement rule by defining a cluster selector using either `key:value` or by providing `matchExpressions`, `matchLabels`, or both, with appropriate values. See `placementRulePath` to specify an existing file.
```

**Review by dhaiducek — COMMENTED** [review]
Thanks for the updates! I have one minor update that might help clarify.

**Comment by dockerymick** [comment]
Thanks @dhaiducek ! i accepted your suggestion. Approving this PR and merging

**Review by dockerymick — APPROVED** [review]
Looks great! Thanks for the changes !

**Pull request merged**